### PR TITLE
feat: add missing rules

### PR DESCRIPTION
**Pull request by bietkul**
- This PR adds`search_settings`, `replace_search_query` and `add_facet` rules.

### CODE REVIEW

File: content/docs/search/rules.md [hunk 1]
```diff
@@ -91,51 +91,97 @@ Here are some example conditions:
...
+
+
 ### Remove Words
 
 Removing words is the progressive loosening of query constraints to include more results when none are initially found.
 
 For example, imagine an online smartphone shop that sold a limited inventory of iPhones in only 16GB and 32GB varieties. Users searching for “iphone 5 64gb” would see no results. This is not ideal behavior - it would be far better to show users some iPhone 5 results instead of a blank page.
 
-You can remove multiple words by space separated values. E.g. `iphone samsung`.
+You can remove multiple words by space-separated values. E.g. `iphone samsung`.
```
Comment (siddharthlatest) [inline content/docs/search/rules.md]:
```suggestion
You can remove multiple words by using space separated values. E.g. `iphone samsung`.
```

File: content/docs/search/rules.md [hunk 2]
```diff
@@ -91,51 +91,97 @@ Here are some example conditions:
...
 
-![remove words](https://i.imgur.com/ArDcJRn.png)
+![remove words](https://i.imgur.com/bVM7oFg.png)
 
 ### Replace Words
 
 Rules offers an alternative. You can now replace words instead of adding new ones. For example, if you make `tv` a synonym for `television`, Rules will replace `tv` with `television` so that only `television` is used to search.
 
 ![replace-word](https://i.imgur.com/ps6JH9A.png)
```
Comment (siddharthlatest) [inline content/docs/search/rules.md:165]: @bietkul This needs a new image based on the latest UI.

File: content/docs/search/rules.md [hunk 3]
```diff
@@ -91,51 +91,97 @@ Here are some example conditions:
...
+This action helps you to define the `dataField` and `fieldWeights` for your `search` type of queries. For example, if you want to always set `product_name` as `dataField` irrespective of what user defines in the search request.
+
+> Note: The `dataField` and `fieldWeights` values set by this rule will override the `dataField` and `fieldWeights` values set in the search request.
+
+![search settings](https://i.imgur.com/cE2jXvS.png)
+
 ### Functions
 
-Helps in doing more customization with search or handling edge cases around search relevancy. Functions lets you implement any custom action. Example you want to perform natural language processing on search query.
+Helps in doing more customization with search or handling edge cases around search relevancy. Functions let you implement any custom action. For example you want to perform natural language processing on the search query.
```
Comment (siddharthlatest) [inline content/docs/search/rules.md]:
```suggestion
Helps in doing more customization with search or handling edge cases around search relevancy. Functions let you implement any custom action. For example, you want to perform natural language processing on the search query.
```

**Review by siddharthlatest — CHANGES_REQUESTED** [review]

File: content/docs/search/rules.md (hunk 2 above)
Comment (bietkul) [reply content/docs/search/rules.md:165]: @siddharthlatest There is no change here.

**Review by siddharthlatest — APPROVED** [review]

**Pull request merged**